onnxruntime/core/framework/session_state.cc

```diff
@@ -618,4 +630,154 @@ const std::unordered_set<NodeIndex>* SessionState::GetToBeExecutedNodes(
   return (it != to_be_executed_nodes_.end()) ? &it->second : nullptr;
 }
 
+Status SessionState::CreateSubgraphSessionState() {
+  for (auto& node : graph_.Nodes()) {
+    for (auto& entry : node.GetAttributeNameToMutableSubgraphMap()) {
+      const auto& ep = node.GetExecutionProviderType();
+      if (ep != kCpuExecutionProvider && ep != kCudaExecutionProvider) {
+        // SessionState is only used by ORT EPs
```

Comment:
So SessionState cannot be used by external library based ORT EPs ?

Reply:
Each subgraph in a control flow node has it's own session state instance that is used if ORT executes the subgraph. If a different EP has taken the whole subgraph it wouldn't be using our session state info.